fasteners/threads.scad

```diff
@@ -44,32 +59,140 @@ use <scad-utils/transformations.scad>
 
 
 // ----------------------------------------------------------------------------
-// internal - true = clearances for internal thread (e.g., a nut).
-//            false = clearances for external thread (e.g., a bolt).
-//            (Internal threads should be "cut out" from a solid using
-//            difference()).
-// n_starts - Number of thread starts (e.g., DNA, a "double helix," has
-//            n_starts=2).  See wikipedia Screw_thread.
+// Metric Threads in accordance with ISO 68-1
+// ----------------------------------------------------------------------------
+// diameter  - nominal diameter of thread in mm. (e.g. 10 for an M10 thread)
+// pitch     - distance in mm from the crest of one thread to the next.
+// length    - overall axial length of thread in mm.
+// internal  - true = clearances for internal thread (e.g., a nut).
+//             false = clearances for external thread (e.g., a bolt).
+//             (Internal threads should be "cut out" from a solid using
+//             difference()).
+// clearance - additional clearance to add in mm. Teeth are effectively
+//             shifted inwards (outwards) for external (internal) threads by
+//             this amount.
+// n_starts  - Number of thread starts (e.g., DNA, a "double helix," has
+//             n_starts=2).  See Wikipedia Screw_thread.
 module metric_thread (
     diameter = 8,
```

Comment:
Just a thought but, should we drop default values for `diameter`, `pitch` and `length`. These should be _mandatory_ arguments in my mind.

Reply:
Yes please. I think that was a carryover from a weird era